layers/data_service_layer/python/data_service.py

```diff
@@ -17,11 +18,60 @@ def __init__(self, table_name: str, dynamodb_resource=None):
         self.dynamodb = dynamodb_resource or boto3.resource('dynamodb')
         self.table = self.dynamodb.Table(table_name)
 
-    def get_markers(self) -> List[dict]:
+    def get_markers(self) -> List[LocationMarker]:
         """
         Retrieve all markers from the DynamoDB table.
 
         :return: A list of markers.
         :raises NotImplementedError: Indicates the method is not yet implemented.
         """
-        raise NotImplementedError("This method is not implemented yet.")
+        try: 
+            response = self.table.scan()
+            markers_data = response.get('Items', [])
+            markers = [LocationMarker.from_json(marker_data) for marker_data in markers_data]
+            logger.info(f"Retrieved {len(markers)} markers.")
+            return markers
+        except Exception as e:
+            logger.error(f"Error retrieving markers: {e}")
+            raise
+
+    def add_marker(self, location_marker: LocationMarker, mark_id):
+        """
+        Add a new marker to the DynamoDB table.
+
+        :param markerId: Unique identifier for the location.
+        :param Name: Name of the location.
+        :param Longitude: Longitude coordinate of the location.
+        :param Latitude: Latitude coordinate of the location.
+        :param ImgUrl: URL to an image of the location.
+        :param Tags: Tags associated with the location.
+        :return: The response from DynamoDB.
+        """
+        try:
+            location_marker.set_marker_id(mark_id)
+            marker_data = location_marker.to_json()
+
+            response = self.table.put_item(Item=marker_data)
```

Comment:
@PCBush a minor suggestion, if you agree, is to do 

`response = self.table.put_item(Item= location_marker.to_json())`

to avoid creating `marker_data`

Reply:
Done